demonstrations/tutorial_classical_shadows.py

```diff
@@ -0,0 +1,601 @@
+    """
+    num_snapshots = shadow.shape[0]
+    num_qubits = shadow.shape[1] // 2
+
+    # classical values
+    b_lists = shadow[:, 0:num_qubits]
+    # pauli observable ids
+    obs_lists = shadow[:, num_qubits : 2 * num_qubits]
+
+    # Averaging over snapshot states.
+    shadow_rho = np.zeros((2 ** num_qubits, 2 ** num_qubits), dtype=complex)
+    for i in range(num_snapshots):
+        shadow_rho += snapshot_state(b_lists[i], obs_lists[i])
+
+    return shadow_rho / num_snapshots
+
+def snapshot_state(b_list, obs_list):
+    """
+    Reconstruct a state from a single snapshot in a shadow.
+
+    **Details:**
+    Implements Eq. (S44) from https://arxiv.org/pdf/2002.08953.pdf
+
+    Args:
+        b_list (array): The list of classical outcomes for the snapshot.
+        obs_list (array): Indices for the applied pauli measurement
+    """
+    num_qubits = len(b_list)
+
+    # computational basis states
+    zero_state = np.array([[1, 0], [0, 0]])
+    one_state = np.array([[0, 0], [0, 1]])
+
+    # local qubit unitaries
+    phase_z = np.array([[1, 0], [0, -1j]], dtype=complex)
+    hadamard = qml.Hadamard(0).matrix
+    identity = qml.Identity(0).matrix
+
+    # rotations from computational basis to x,y,z respectively
+    unitaries = [hadamard, hadamard @ phase_z, identity]
+
+    # reconstructing the snapshot state from local pauli measurements
+    rho_snapshot = [1]
+    for i in range(num_qubits):
+        state = zero_state if b_list[i] == 1 else one_state
+        U = unitaries[int(obs_list[i])]
+
+        # applying Eq. (S44)
+        local_rho = 3 * (U.conj().T @ state @ U) - identity
+        rho_snapshot = np.kron(rho_snapshot, local_rho)
+
+    return rho_snapshot
+
+##############################################################################
+# Example: Reconstructing a Bell State
+# ************************************
+# First, we construct a single-shot, ``'default.qubit'`` device and
+# define the ``bell_state_circuit`` QNode to construct and measure a Bell state.
+
+num_qubits = 2
+
+dev = qml.device("default.qubit", wires=num_qubits, shots=1)
+
+# circuit to create a Bell state and measure it in
+# the bases specified by the 'observable' keyword argument.
+@qml.qnode(device=dev)
+def bell_state_circuit(params, wires, **kwargs):
+    observables = kwargs.pop("observable")
+
+    qml.Hadamard(0)
+    qml.CNOT(wires=[0, 1])
+
+    return [qml.expval(o) for o in observables]
+
+##############################################################################
+# Then, construct a classical shadow consisting of 1000 snapshots.
+
+num_snapshots = 1000
+params = []
+
+shadow = calculate_classical_shadow(bell_state_circuit, params, num_snapshots, num_qubits)
+shadow
+
+##############################################################################
+# To reconstruct the Bell state use ``shadow_state_reconstruction``.
+
+shadow_state = shadow_state_reconstruction(shadow)
+shadow_state
```

Comment:
```suggestion
print(np.round(shadow_state, decimals=6))
```
(Just helps with readability of the output)